test_plans/oneapi/ext_oneapi_current_device.asciidoc

```diff
@@ -0,0 +1,35 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for ext oneapi current device
+
+This is a test plan for the device selector as described in SYCL 2020 revision 9 specification.
+
+Estimated development time is one day.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are not backend-specific and are performed for any SYCL backend.
+
+=== Device coverage
+
+All tests construct a test device for which conformance is assessed. All of the tests described below are performed only on the default device that
+can be selected on the CTS command line.
+
+== Tests
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_CURRENT_DEVICE` so they can be skipped
+if feature is not supported.
+
+=== Functions `get_current_device` and `set_current_device`
+
+1. Get default device name from environment variable. Call `get_current_device()` but do not call `set_current_device`.
+Compare value stored in env variable and returned by `get_current_device()` device names.
+2. Call `set_current_device(d)` then call `get_current_device()` from the same thread.
+Compare return value of `get_current_device()` with `d`.
+3. Create multiple threads. Each thread calls the following in a loop: `set_current_device(d)` followed by `get_current_device()`. Each thread uses a different device `d` (if the system has more than one device). 
+Make sure the call to get_device returns the same device that was set in `set_current_device(d)`. This tests that each thread is keeping a separate copy of the "current device".
```

Comment:
I was thinking about this more.  A better test would be:

* Thread 1 does:
  * Call `set_current_device(d1)`
  * Signal thread 2
  * Wait for thread 2 to signal
  * Call `get_current_device()` and make sure it returns `d1`.
* Thread 2 does:
  * Call `set_current_device(d2)`
  * Signal thread 1
  * Wait for thread 1 to signal
  * Call `get_current_device()` and make sure it returns `d2`.
 
Here `d1` and `d2` should be different devices, if the system has more than one device.  There is no need to run this in a loop.